api/gateway/apimiddleware/api_middleware.go

```diff
@@ -111,7 +111,7 @@ func (m *ApiProxyMiddleware) WithMiddleware(path string) http.HandlerFunc {
 			}
 		}
 
-		if req.Method == "DELETE" {
+		if req.Method == "DELETE" && req.Body != http.NoBody {
```

Comment:
Why this change?

Reply:
my delete function has no body sent, if i don't have this statement then it tries to marshal a body that doesn't exist into another entity that doesn't exist.